api/omnitransaction.py

```diff
@@ -0,0 +1,124 @@
+import urlparse
+import os, sys, re, random, pybitcointools, bitcoinrpc, math
+from decimal import Decimal
+from msc_apps import *
+from blockchain_utils import *
+import config
+
+class OmniTransaction:
+    confirm_target=6
+    HEXSPACE_SECOND='21'
+    mainnet_exodus_address='1EXoDusjGwvnjZUyKkxZ4UHEf77z6A5S4P'
+    testnet_exodus_address='mpexoDuSkGGqvqrkrjiFng38QPkJQVFyqv'
+
+    def __init__(self,tx_type,form):
+        self.conn = getRPCconn()
+        self.testnet = False
+        self.magicbyte = 0
+        self.exodus_address=self.mainnet_exodus_address
+
+        if 'testnet' in form and ( form['testnet'] in ['true', 'True'] ):
+            self.testnet =True
+            self.magicbyte = 111
+            self.exodus_address=self.testnet_exodus_address
+
+        try:
+          if config.D_PUBKEY and ( 'donate' in form ) and ( form['donate'] in ['true', 'True'] ):
+            print "We're Donating to pubkey for: "+pybitcointools.pubkey_to_address(config.D_PUBKEY)
+            self.pubkey = config.D_PUBKEY
+          else:
+            print "not donating"
+            self.pubkey = form['pubkey']
+        except NameError, e:
+          print e
+          self.pubkey = form['pubkey']
+        self.fee = estimateFee(self.confirm_target)['result']
+        self.rawdata = form
+        self.tx_type = tx_type
+
+    def get_unsigned(self):
+        # get payload
+        payload = self.__generate_payload()
+        # Add exodous output
+        rawtx = createrawtx_reference(self.exodus_address)['result']
```

Comment:
This is not needed. In case of OP_RETURN, we no longer need the Exodus address, and when adding multisig payloads with `omni_createrawtx_multisig`, then the Exodus output is added.